client/verta/tests/unit_tests/test_deployed_model.py

```diff
@@ -0,0 +1,346 @@
+
+
+def test_predict_http_defaults_200(mocked_responses) -> None:
+    """ Calling predict with the default settings and getting a 200 response returns the response as expected. """
+    mocked_responses.post(
+        PREDICTION_URL,
+        json={"test_key": "test_val"},
+        status=200,
+        headers={'verta-request-id': 'hereISaTESTidFROMtheUSER'},
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    prediction_response = dm.predict(x=['test_prediction'])
+    assert prediction_response == {"test_key": "test_val"}
+
+
+def test_predict_http_defaults_404_retry_error(mocked_responses) -> None:
+    """ Calling predict with the default settings and getting a 404 results in retries being exhausted. """
+    mocked_responses.post(
+        PREDICTION_URL,
+        json={},
+        status=404,
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    with pytest.raises(RetryError):
+        dm.predict(x=['test_prediction'])
+
+
+def test_predict_http_defaults_429_retry_error(mocked_responses) -> None:
+    """ Calling predict with the default settings and getting a 429 results in retries being exhausted. """
+    mocked_responses.post(
+        PREDICTION_URL,
+        json={},
+        status=429,
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    with pytest.raises(RetryError):
+        dm.predict(x=['test_prediction'])
+
+
+def test_predict_http_defaults_status_not_in_retry(mocked_responses) -> None:
+    """ Verify that calling predict with the default settings and getting a response not in `status_forcelist`
+     does not result in retries. """
+    mocked_responses.post(
+        PREDICTION_URL,
+        headers={'verta-request-id': 'hereISaTESTidFROMtheUSER'},
+        json={},
+        status=999,
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    dm.predict(x=['test_prediction'])
+    mocked_responses.assert_call_count(PREDICTION_URL, 1)
+
+
+def test_predict_http_default_max_retry_observed(mocked_responses) -> None:
+    """ Calling predict with the default settings and getting a 429 results in retries being exhausted. """
+    mocked_responses.post(
+        PREDICTION_URL,
+        json={},
+        status=429,
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    try:
+        dm.predict(x=['test_prediction'])
+    except(RetryError):
```

Comment:
Could we use `with pytest.raises(RetryError):` here, too? or does `mocked_responses.assert_call_count()` preclude that?

Reply:
@ewagner-verta This didn't quite address my question!

I'm asking if we could use `with pytest.raises(RetryError)` (not `except RetryError`) because we should be explicitly checking that a `RetryError` is raised.

If we simply use `except` rather than `with pytest.raises()`, then if there is a bug in `dm.predict()` that ends up _not_ raising a `RetryError`, this test would happily pass none the wiser (and similarly for the other tests using `except RetryError`).